CMakeLists.txt

```diff
@@ -566,9 +566,7 @@ if (CMAKE_COMPILER_IS_GNUCC)
   set(BASE_C_FLAGS "${BASE_C_FLAGS}")
 endif ()
 
-if (CMAKE_COMPILER_IS_GNUCC OR APPLE)
-  set(BASE_CXX_FLAGS "${BASE_CXX_FLAGS} -std=c++17")
-endif ()
+set(BASE_CXX_FLAGS "${BASE_CXX_FLAGS} -std=c++17")
```

Comment:
For MSVC correct spelling is "-std:c++17"
-std=c++17 will just produce "1>cl : command line warning D9002: ignoring unknown option '-std=c++17'"

So maybe just leave this condition as is but add flag below where MSVC options are set

Reply:
done